Tasks/VsTestV2/runvstest.ts

```diff
@@ -78,6 +79,39 @@ async function execute() {
     }
 }
 
+function isHydraFlowToBeEnabled(inputDataContract: InputDataContract) {
+    if ((inputDataContract.ServerType && inputDataContract.ServerType.toLowerCase() === 'hosted')) {
+
+        tl.debug('Enabling Hydra flow since serverType is hosted.');
+        return true;
+    }
+
+    if (tl.getVariable('Force_Hydra') && tl.getVariable('Force_Hydra').toLowerCase() === 'true') {
+
+        tl.debug('Enabling Hydra flow since Force_Hydra build variable is set to true.');
+        return true;
+    }
+
+    if (inputDataContract.TestReportingSettings && inputDataContract.TestReportingSettings.ExecutionStatusSettings
+        && !utils.Helper.isNullEmptyOrUndefined(inputDataContract.TestReportingSettings.ExecutionStatusSettings.ActionOnThresholdNotMet)
+        && inputDataContract.TestReportingSettings.ExecutionStatusSettings.ActionOnThresholdNotMet !== 'donothing') {
+
+        tl.debug('Enabling Hydra flow since the minimum test executed feature is being used.');
+        return true;
+    }
+
+    if (inputDataContract.TestReportingSettings
+        && !utils.Helper.isNullEmptyOrUndefined(inputDataContract.TestReportingSettings.TestResultsDirectory)
+        && inputDataContract.TestReportingSettings.TestResultsDirectory.toLowerCase()
+            !== path.join(tl.getVariable('Agent.TempDirectory'), 'TestResults').toLowerCase()) {
+
+        tl.debug('Enabling Hydra flow since the override results directory feature is being used.');
+        return true;
+    }
+
+    return false;
```

Comment:
consider wrapping this in try / catch and return false if there is any errors with any of these condition checks

Reply:
fair point